### PR TITLE
scan-build 278 (new formula)

### PR DESCRIPTION
**Pull request by mgrebenets**
A follow up to [this pull request](https://github.com/caskroom/homebrew-cask/pull/17456) to Homebrew/Cask.

### All Submissions:

- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew/blob/master/.github/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew/pulls) for the same update/change?

### New Formulae Submissions:

- [x] Does your submission pass
`brew audit --strict --online <formula>` (where `<formula>` is the name of the formula you're submitting)?
- [x] Have you built your formula locally prior to submission with `brew install <formula>`?

### CODE REVIEW

File: Library/Formula/scan-build.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+  homepage "http://clang-analyzer.llvm.org/scan-build.html"
+  url "http://clang-analyzer.llvm.org/downloads/checker-278.tar.bz2"
+  sha256 "f7180d530eceafc006f30461895ac23033b3b432c41363a185f2234535a09860"
+
+  def install
+    prefix.install Dir["*"]
+  end
+
+  test do
+    system "#{bin}/scan-build"
```
Comment (mikemcquaid) [inline Library/Formula/scan-build.rb]: Test failing: http://bot.brew.sh/job/Homebrew%20Pull%20Requests/43102/version=el_capitan/testReport/junit/brew-test-bot/el_capitan/test_scan_build/

Comment (mikemcquaid) [inline Library/Formula/scan-build.rb]: Can this test be modified to do something more substantial than e.g. `--version` or `--help`? See `cmake.rb` for an example of an application formula with a good test and `tinyxml2.rb` for an example of a library formula with a good test. Thanks!

**Comment by mikemcquaid** [comment]
This is currently already installed by `llvm --with-clang`.

File: Library/Formula/scan-build.rb (hunk 1 above)
Comment (mgrebenets) [inline Library/Formula/scan-build.rb]: Modified to use `--help`.

**Comment by mgrebenets** [comment]
True, just got it installed. It took a while though and I had to do `brew link llvm --force` with a certain warning saying that I should be careful.

What are the standards? Is it fine to have formulas that might conflict? 
We need to provision our CI Mac boxes with `scan-build` only and we don't need the whole `llvm --with-clang` installed. 

If this formula cannot be merged here and in Cask(https://github.com/caskroom/homebrew-cask/pull/17456) is using a tap my only option?

**Comment by mgrebenets** [comment]
Another thing I've just discovered.

I need to use `scan-build` for iOS projects, so I want `xcodebuild` to use `clang` bundled up with my Xcode installation. Once I install custom `llvm` running `xcodebuild` fails with errors like this one:

```
clang-3.6: error: unknown argument: '-gmodules'
clang-3.6: error: unknown argument: '-fapplication-extension'
Command /usr/local/Cellar/llvm/3.6.2/bin/clang-3.6 failed with exit code 1
```

So in my case I really can't install custom `llvm` at all.

**Comment by mikemcquaid** [comment]
It feels like making this a tap is a better idea. @apjanke was talking about making `with-clang` the default option for LLVM; we'd accept a PR that did that instead?

**Comment by DomT4** [comment]
> @apjanke was talking about making with-clang the default option for LLVM; we'd accept a PR that did that instead?

I believe his PR on that may still be open. Some things actively broke because of the change AFAIK which is the hold-up.

**Comment by mgrebenets** [comment]
The `llvm` doesn't fit me anyways. I have to stick with `xcodebuild` and Apple LLVM version. Closing this one will create a tap. Thank.

**Comment by apjanke** [comment]
> > @apjanke was talking about making with-clang the default option for LLVM; we'd accept a PR that did that instead?
>
> I believe his PR on that may still be open. Some things actively broke because of the change AFAIK which is the hold-up.

Oh, yep: it's #48462. Sorry; that PR is stale: it can't be merged as is (because it breaks builds/tests) and I don't know how to fix it right now. We should probably just close that PR and maybe come back to it later. It's clear there are some problems there, and I'm not qualified to sort them out.

I still think enabling Clang, RTTI, and maybe `libcxx` by default are probably a good idea for Homebrew (especially considering the bottling implications), if feasible, but I wasn't able to make it work there and don't see what the next logical step is or have time to pursue it at the moment. Esp because I'm not an LLVM expert and don't really understand the clang/C++/libc++/C++11/etc interactions, and I understand they are *cough cough* non-trivial.

**Pull request closed without merging**